fedimint-client/src/meta.rs

```diff
@@ -0,0 +1,351 @@
+                    },
+                )
+            })
+            .collect())
+    }
+}
+
+pub const LEGACY_META_OVERRIDE_DEFAULT_SOURCE_ID: u64 = 1;
+pub const LEGACY_META_OVERRIDE_DEFAULT_PRIORITY: u64 = 1;
+
+#[derive(Debug, Clone)]
+pub struct LegacyMetaOverrideUrlSource {
+    client: reqwest::Client,
+    field_name: Cow<'static, str>,
+    priority: u64,
+    source_id: u64,
+}
+
+impl Default for LegacyMetaOverrideUrlSource {
+    fn default() -> Self {
+        Self {
+            client: Default::default(),
+            field_name: "meta_override_url".into(),
+            priority: LEGACY_CLIENT_CONFIG_PRIORITY,
+            source_id: LEGACY_CLIENT_CONFIG_SOURCE_ID,
+        }
+    }
+}
+impl LegacyMetaOverrideUrlSource {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn custom(field_name: impl Into<Cow<'static, str>>, source_id: u64, priority: u64) -> Self {
+        Self {
+            client: Default::default(),
+            field_name: field_name.into(),
+            priority,
+            source_id,
+        }
+    }
+}
+
+#[apply(async_trait_maybe_send!)]
+impl MetaSource for LegacyMetaOverrideUrlSource {
+    fn source_id(&self) -> u64 {
+        self.source_id
+    }
+
+    async fn wait_for_update(&self) {
+        // update every 10 minutes
+        fedimint_core::task::sleep(Duration::from_secs(10 * 60)).await
+    }
+
+    async fn fetch(
+        &self,
+        client: &Client,
+    ) -> anyhow::Result<BTreeMap<MetaFieldName, MetaFieldValue>> {
+        let Some(url) = client.config().meta::<String>(&self.field_name)? else {
+            return Ok(BTreeMap::new());
+        };
+        let response = self
+            .client
+            .get(url)
+            .send()
+            .await
+            .context("Meta override source could not be fetched")?;
+
+        debug!("Meta override source returned status: {response:?}");
+
+        if response.status() != reqwest::StatusCode::OK {
+            bail!(
+                "Meta override request returned non-OK status code: {}",
+                response.status()
+            );
+        }
+
+        let meta_fields_raw = response
+            .json::<BTreeMap<String, serde_json::Value>>()
+            .await
+            .context("Meta override could not be parsed as JSON")?;
+
+        let meta_fields = meta_fields_raw
+            .into_iter()
+            .filter_map(|(key, value)| match value.as_str() {
+                Some(value_str) => Some((
+                    MetaFieldName(key),
+                    MetaFieldValue {
+                        value: value_str.to_owned(),
+                        revision: None,
+                        priority: self.priority,
+                    },
+                )),
+                None => {
+                    warn!("Meta override map contained non-string key: {key}, ignoring");
+                    None
+                }
+            })
+            .collect::<BTreeMap<_, _>>();
```

Comment:
I think you forgot to take into account here that there can be multiple federation entries in a `meta.json`, thus we have to first get the entry corresponding to our federation (might be my mistake if you copied it, ran into it with `fedimint-observer` too).

Reply:
fixed it